qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -59,182 +57,75 @@ class CurveAnalysis(BaseAnalysis, ABC):
-                    SeriesDef(
-                        fit_func=lambda x, p0, p1, p2:
-                            exponential_decay(x, amp=p0, lamb=p1, baseline=p2),
-                    ),
-                ]
-
-                __fixed_parameters__ = ["p1"]
-
-        You can add arbitrary number of parameters to the class variable
-        ``__fixed_parameters__`` from the fit function arguments.
-        This parameter should be defined with the fit functions otherwise the analysis
-        instance cannot be created. In above example, parameter ``p1`` should be also
-        defined in the analysis options. This parameter will be excluded from the fit parameters
-        and thus will not appear in the analysis result.
-
-    Notes:
-        This CurveAnalysis class provides several private methods that subclasses can override.
-
-        - Customize pre-data processing:
-            Override :meth:`~self._format_data`. For example, here you can apply smoothing
-            to y values, remove outlier, or apply filter function to the data.
-            By default, data is sorted by x values and the measured values at the same
-            x value are averaged.
-
-        - Create extra data from fit result:
-            Override :meth:`~self._extra_database_entry`. You need to return a list of
-            :class:`~qiskit_experiments.framework.analysis_result_data.AnalysisResultData`
-            object. This returns an empty list by default.
-
-        - Customize fit quality evaluation:
-            Override :meth:`~self._evaluate_quality`. This value will be shown in the
-            database. You can determine the quality represented by the predefined string
-            "good" or "bad" based on fit result,
-            such as parameter uncertainty and reduced chi-squared value.
-            This returns ``None`` by default. This means evaluation is not performed.
-
-        - Customize fitting options:
-            Override :meth:`~self._generate_fit_guesses`. For example, here you can
-            calculate initial guess from experiment data and setup fitter options.
-
-        See docstring of each method for more details.
-
-        Note that other private methods are not expected to be overridden.
-        If you forcibly override these methods, the behavior of analysis logic is not well tested
-        and we cannot guarantee it works as expected (you may suffer from bugs).
-        Instead, you can open an issue in qiskit-experiment github to upgrade this class
-        with proper unittest framework.
-
-        https://github.com/Qiskit/qiskit-experiments/issues
-    """
+        super().__init_subclass__(**kwargs)
 
-    #: List[SeriesDef]: List of mapping representing a data series
-    __series__ = list()
+        # Validate if all fixed parameter names are defined in the fit model
+        if cls.__fixed_parameters__:
+            all_params = set(itertools.chain.from_iterable(s.signature for s in cls.__series__))
```

Comment:
This is for backward compatibility, i.e. if one specifies non-existing parameter in the `__fixed_parameters__` it should raises an error. Since curve signature is a list, this flattens list of list to a flat set. This is equivalent to 
```python
all_params = []
for s in cls.__series__:
    all_params.extend(s.signature)
all_params = set(all_params)
```
but mush simpler (I didn't compare the performance).